assets/js/base/components/cart-checkout/totals/totals-footer-item/index.js

```diff
@@ -23,6 +26,7 @@ const TotalsFooterItem = ( { currency, values } ) => {
 			label={ __( 'Total', 'woo-gutenberg-products-block' ) }
 			value={ parseInt( totalPrice, 10 ) }
 			description={
+				TAXES_ENABLED &&
```

Comment:
I'm a bit worried about mixing conditions in the render return. I've run into this issue before in which having 3 conditions results in 9 implicit possibilities that you need to juggle in your mind.

In this case, `TAXES_ENABLED` and `DISPLAY_CART_PRICES_INCLUDING_TAX` are related, the second has no meaning if the first is disabled.

I have two options:

We add an extra condition outside the function called `DISPLAY_TAXES` that is false if taxes are disabled and assumes the value of `DISPLAY_CART_PRICES_INCLUDING_TAX` otherwise:

```js
const SHOW_TAXES = TAXES_ENABLED && DISPLAY_CART_PRICES_INCLUDING_TAX;
```

Another option is to disable `DISPLAY_CART_PRICES_INCLUDING_TAX` at the definition level, so that it always returns false if `TAXES_ENABLED` is false.
This option can have certain side effects that we can't see, but because of they are related (controlling the option of `DISPLAY_CART_PRICES_INCLUDING_TAX` is not possible if taxes are disabled) I think it's safe to use that option.

Reply:
Hmm, interesting point @senadir . Could also factor out the tax summary to a mini-component and add the logic there, use early return to make it super simple. Something like:

```js
function TaxSummaryLine( { currency, totaltqx, } ) {
  if ( ! TAXES_ENABLED || ! DISPLAY_CART_PRICES_INCLUDING_TAX ) {
    return null;
  }

  return ( 
    <p className="wc-block-components-totals-footer-item-tax">
      { createInterpolateElement(
        __(
          'Including <TaxAmount/> in taxes',
          'woo-gutenberg-products-block'
        ),
        {
          TaxAmount: (
            <FormattedMonetaryAmount
              className="wc-block-components-totals-footer-item-tax-value"
              currency={ currency }
              displayType="text"
              value={ parseInt( totalTax, 10 ) }
            />
          ),
        }
      ) }
    </p>
  );
}
```

Reply:
We can't change the definition of `DISPLAY_CART_PRICES_INCLUDING_TAX` to depend on `TAXES_ENABLED`. Although it sounds reasonable here this would make those two values behave differently to their serverside counterparts - that is an unnecessary complication, and it has the potential to generate bugs in the future.

I do like the idea of extracting that bit of logic outside into a `SHOW_TAXES` just outside the `return` statement. I would not want to put it further - Is that correct @senadir ? That extraction makes sense also because it gives a direct name to the outcome of the logical condition.

As for extraction into a separate component - I don't know much about the architecture of the project yet to even remotely know if that is a correct approach - it feels like overkill though. Especially the name makes things a bit confusing. This tax line would be just applicable to CART and CHECKOUT and not any other place.

Reply:
> We can't change the definition of DISPLAY_CART_PRICES_INCLUDING_TAX to depend on TAXES_ENABLED. Although it sounds reasonable here this would make those two values behave differently to their serverside counterparts - that is an unnecessary complication, and it has the potential to generate bugs in the future.

In any other case, I would agree with this. But for this particular case, options like should reflect the default value when disabled, as for bugs, I'm okay with having them and generating them, there should be no case when in which `TAXES_ENABLED` is `false` and `DISPLAY_CART_PRICES_INCLUDING_TAX` because it's impossible to reach that case intentionally.

However, I'm not trying to block this PR, just to shed light on how complex mixing settings can get.

> I do like the idea of extracting that bit of logic outside into a SHOW_TAXES just outside the return statement. I would not want to put it further - Is that correct @senadir ?

Yes, generally we would move it to the top level of that file, outside any function, so that it's not regenerated on each render.

> As for extraction into a separate component - I don't know much about the architecture of the project yet to even remotely know if that is a correct approach - it feels like overkill though. Especially the name makes things a bit confusing. This tax line would be just applicable to CART and CHECKOUT and not any other place.

Having it as a separate component is completely okay, we have a couple of packages for cart and checkout components only.
Cart and Checkout are huge; if we leave them in a single file, they will reach thousands of lines.

But in this case, I'm between. It will provide more readability for this component.
You can abstract the component and leave it within that file.
The general goal is to break components to the lowest possible function, this is useful for several things, like extensibility, reusability and performance.
However, I don't feel strongly about this, if usually try not to abstract unless there's a problem, so don't feel like this should block you. You can merge it anyway.